.github/workflows/code-test-coverage.yml

```diff
@@ -37,9 +42,29 @@ jobs:
       - name: Generate full test coverage report using go-acc
         run: make test:coverage
 
-      - name: Upload coverage to Codecov
+      - name: Upload coverage to Codecov without token, retry on failure
+        env: 
+          codecov_secret: ${{ secrets.CODECOV_TOKEN }}
+        if: env.codecov_secret == ''
+        uses: Wandalen/wretry.action@v1.0.36
```

Comment:
todo: Can we please at least use a more popular dependency *if* we are going to introduce one?  The one currently on this line only has 35 stars, whereas this one has 245 and was built at/for a well known company whereas `wandalen`s appears to be purely a personal project.

Reply:
Which one are you recommending? I just saw many people use this one to fix the code cov flakey stuff. Happy to look into another one that might be more reputable.

Reply:
lol, I forgot to link it. I had it open and ready for you sorry. This is the one I meant to suggest: https://github.com/marketplace/actions/retry-step

Reply:
It seems to be intended to replace the run step commands not action uses. LMK if you see an example for retries upon an action use.

Reply:
More info:
https://stackoverflow.com/a/71583518

Reply:
You should be able to run the code cov action as a command no?

I'm just very uncomfortable introducing such an unused personal project into our CI system, and I think if I had to choose between using that action, or having a flaky code cov action, I would probably go for the flaky code cov action.

Reply:
I would rather use the official codecov github action than the shell command as it takes into account some built-in github metadata when it uploads the data to codcov.

I disagree that it is `an unused personal project` it has thousands of uses already: https://github.com/search?q=Wandalen%2Fwretry.action+path%3A.github%2Fworkflows%2F+language%3AYAML&type=Code&s=indexed&o=desc

Some of the users are like shopify, open-metadata, Isar Database, Airbyte, Immich

We shouldn't not solve the problem just because we have to use a 3rd party action. We already use a lot of 3rd party actions, even when we weren't open-sourced.

Reply:
Also keep in mind, this will mostly be a fallback case when a token isn't available (ASFAIK it will always be there). So majority of times this won't even be hit or used once the rework of this action is done.

Reply:
I think it's pretty safe to use the one Shahzad has at the moment. Specially that it's set on a version so future changes won't affect us unless we update without checking what the changes are.